Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -331,6 +345,148 @@ public LibraryDependenciesResult(@Nullable final String[] libraries,
     }
   }
 
+  private void configureImportedLibraries(@NotNull final Collection<String> rootsToAddToLib) {
+    runInEventDispatchThread(new Runnable() {
+      @Override
+      public void run() {
+        ApplicationManager.getApplication().runWriteAction(new Runnable() {
+          public void run() {
+            doConfigureImportedLibraries(rootsToAddToLib);
+          }
+        });
+      }
+    });
+  }
+
+  private static void runInEventDispatchThread(final Runnable runnable) {
+    try {
+      if (SwingUtilities.isEventDispatchThread()) {
+        runnable.run();
+      }
+      else {
+        ApplicationManager.getApplication().invokeAndWait(new Runnable() {
+          public void run() {
+            runnable.run();
+          }
+        }, ModalityState.defaultModalityState());
+      }
+    }
+    catch (Exception e) {
+      LOG.warn(e);
+    }
+  }
+
+  private void doConfigureImportedLibraries(@NotNull final Collection<String> libraries) {
+    for (final Project project : myRootsHandler.getTrackedProjects()) {
```

Comment:
Good question.  1 project should be sufficient, but it is nice to have more than one for other kinds of projects, which is why I recommend ultimate.